lib/io/hyped_spi/hyped_spi_derive/src/lib.rs

```diff
@@ -13,7 +13,53 @@ pub fn hyped_spi_derive(input: TokenStream) -> TokenStream {
 fn impl_hyped_spi(ast: &syn::DeriveInput) -> TokenStream {
     let name = &ast.ident;
     let generics: &Generics = &ast.generics;
-    let (_impl_generics, _ty_generics, _) = generics.split_for_impl();
-    let gen = quote! {};
+    let (impl_generics, ty_generics, _) = generics.split_for_impl();
+    let gen = quote! {
+        impl #impl_generics HypedSpi for #name #ty_generics{
+            fn read(&mut self, words: &mut [u8]) -> Result<(), SpiError> {
+                match self.spi.blocking_read(words) {
+                    Ok(_) => Ok(()),
+                    Err(e) => Err(match e {
+                        spi::Error::Framing => SpiError::Framing,
+                        spi::Error::Crc => SpiError::Crc,
+                        spi::Error::ModeFault => SpiError::ModeFault,
+                        spi::Error::Overrun => SpiError::Overrun,
```

Comment:
would be nicer to have this mapping be one separate function used by all three `read`, `write` and `transfer_in_place`